test/e2e/helpers.go

```diff
@@ -823,3 +829,94 @@ func getPodLogs(ctx context.Context, clientset *kubernetes.Clientset, pod corev1
 	}
 	return b.String()
 }
+
+func WaitForClusterKubeconfigSecret(ctx context.Context, input clusterctl.ApplyClusterTemplateAndWaitInput) {
+	By("Waiting for workload cluster kubeconfig secret")
+	Eventually(func() error {
+		c := input.ClusterProxy.GetClient()
+		secret := &corev1.Secret{}
+		key := client.ObjectKey{
+			Name:      fmt.Sprintf("%s-kubeconfig", input.ConfigCluster.ClusterName),
+			Namespace: input.ConfigCluster.Namespace,
+		}
+		return c.Get(ctx, key, secret)
+	}, input.WaitForControlPlaneIntervals...).Should(Succeed())
+}
+
+func WaitForRunningNodes(ctx context.Context, input clusterctl.ApplyClusterTemplateAndWaitInput) {
```

Comment:
same here and for running pods below. As much as possible we should share e2e logic with the other providers in the CAPI framework instead of implementing our own logic.

Reply:
I was able to rely upon pre-existing cluster-api conveniences for a part of this, but there is the "ensure n pods are Running" that I had to implement here.